src/components/button/Button.js

```diff
@@ -0,0 +1,44 @@
+import './styles.css';
+
+class Button {
+  constructor({ label = '', onClick = undefined, type = 'default', imageSrc = null, id, width = 'auto' }) {
+    this.is = id;
+    this.label = label;
+    this.onClick = onClick;
+    this.type = type;
+    this.imageSrc = imageSrc;
+    this.width = width;
+  }
+
+  render() {
+    const btn = document.createElement(`button`);
+    const img = document.createElement(`img`);
+    img.src = this.imageSrc;
+    btn.appendChild(img);
+    btn.innerText = this.label;
+
+    if (this.type === 'default') {
```

Comment:
if (this.type === 'default') {
      btn.classList.add('default');
    }
    elseif (this.type === 'hof') {
      btn.classList.add('hof');
    }
    else (this.type === 'ptg') {
      btn.classList.add('ptg');
    }

or you can type 
this.type === 'default' ? btn.classList.add('default') : (btn.classList.add('hof') ? btn.className = 'hof': btn.classList.add('ptg'))

this should work also
https://developer.mozilla.org/pl/docs/Web/JavaScript/Reference/Operators/Conditional_Operator